pkg/controller/config/config_controller.go

```diff
@@ -152,34 +165,46 @@ func (r *ReconcileConfig) Reconcile(req reconcile.Request) (reconcile.Result, er
 		return reconcile.Result{}, err
 	}
 
-	if isUpToDate(cfg) {
+	if stateCheck(cfg, op.InstalledStatus) {
 		log.Info("skipping installation, resource already up to date")
 		return reconcile.Result{}, nil
 	}
-
+	if stateCheck(cfg, op.ValidatedPipeline) {
```

Comment:
:+1: for idiomatic status checks.

Additionally wouldn't be nice flow would be organic?
- [ ] installing pipelines if not installed
- [ ] validate pipeline if installing
- [ ] install add on if pipeline is validated

Reply:
I was thinking of a state machine like this .. 

```
switch cfg.Status() {
case ErrorStatus: return applyPipeline()
case  UnknownStatus: return applyPipeline()
case  ApplyingPipeline: return applyPipeline()
case AppliedPipeline: return validatePipeline()
case ValidatingPipeline: return validatePipeline()
case ValidatedPipeline: return applyAddons()
case AppliedAddons: return markInstalled() 
case InstalledState:     return validateUptoDate() // checks uptodate ? including uuid matching
}
```

Reply:
IMHO; this state machine should suffice ... 

```
   switch cfg.InstallStatus() {
    case op.ErrorStatus:
      return r.applyPipeline(req, cfg)
    case op.EmptyStatus:
      return r.applyPipeline(req, cfg)
    case op.AppliedPipeline:
      return r.validatePipeline(req, cfg)
    case op.ValidatedPipeline:
      return r.applyAddons(req, cfg)
    case op.InstalledStatus:
      return r.validateVersion(req, cfg)
```